analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.CSharp11.cs

```diff
@@ -33,6 +33,42 @@ void ListPattern(List<int[]> list)
 
             }
         }
+
+        void OtherIsPatterns(List<string> strings, List<Tuple<string, int>> tuples)
+        {
+            const string target = "42";
+
+            foreach (var s in strings) // Noncompliant
+            {
+                if (s is target) // Secondary 
+                {
+                    Console.WriteLine("Pattern match successful");
+                }
+            }
+
+            foreach (var s in strings) // Compliant, do not raise on VarPattern in IsPattern
+            {
+                if (s is var s2)
+                {
+                    Console.WriteLine("Pattern match successful");
+                }
+            }
+
+            foreach (var s in strings) // Compliant, do not raise on SingleVariableDeclaration in IsPattern
+            {
+                if (s is { Length: 42 } str) 
+                {
+                    Console.WriteLine("Pattern match successful");
+                }
+            }
+
+            foreach (var t in tuples) // Compliant, do not raise on ParenthesizedVariableDeclaration in IsPattern
+            {
+                if (t is var (t1, t2)) 
+                {
+                    Console.WriteLine("Pattern match successful");
+                }
+            }
+        }
```

Comment:
If I am not mistaken these cases are not C# 11 related. please move them to their proper test case file.